policy/policy-authority.go

```diff
@@ -139,54 +122,55 @@ func (e NonPublicError) Error() string         { return "Name does not end in a
 //  * MUST NOT be a label-wise suffix match for a name on the black list,
 //    where comparison is case-independent (normalized to lower case)
 //
-// XXX: Is there any need for this method to be constant-time?  We're
-//      going to refuse to issue anyway, but timing could leak whether
-//      names are on the blacklist.
+// If WillingToIssue returns an error, it will be of type MalformedRequestError.
 func (pa PolicyAuthorityImpl) WillingToIssue(id core.AcmeIdentifier, regID int64) error {
 	if id.Type != core.IdentifierDNS {
-		return InvalidIdentifierError{}
+		return errInvalidIdentifier
 	}
 	domain := id.Value
-
-	for _, ch := range []byte(domain) {
-		if !isDNSCharacter(ch) {
-			return SyntaxError{}
-		}
+	if domain == "" {
```

Comment:
Removing the byte-by-byte check for DNS characters seems (1) dangerous, and (2) out of scope for this change.  Can you clarify why this makes sense?  IIRC, we added that check after AGL expressed some concerns about using string comparison and regexes in light of things like null characters (which have been a problem for certs in the past).